### PR TITLE
Replace twisted with asyncio

### PR DESCRIPTION
**Pull request by egbertbouman**

### CODE REVIEW

File: ipv8/REST/base_endpoint.py [hunk 1]
```diff
@@ -1,62 +1,46 @@
 from __future__ import absolute_import
 
-from twisted.web import resource
-from twisted.web.resource import _computeAllowedMethods
+import json
```
Comment (qstokkink) [inline ipv8/REST/base_endpoint.py]: Why take out the `json_util` here?

**Comment by qstokkink** [comment]
For future cherry-picking reference, PRs merged since opening this PR:

 - [ ] #600

**Comment by egbertbouman** [comment]
retest this please

**Comment by egbertbouman** [comment]
retest this please

**Comment by egbertbouman** [comment]
retest this please

File: ipv8/attestation/wallet/irmaexact/enroll_script.py [hunk 2]
```diff
@@ -155,8 +149,8 @@ def b64_to_int(s):
 
 
 def str_to_int(s):
-    if isinstance(s, six.text_type):
-        s = b''.join(six.int2byte(ord(c)) for c in s)
+    if isinstance(s, srt):
```
Comment (qstokkink) [inline ipv8/attestation/wallet/irmaexact/enroll_script.py]: Shouldn't this be `str`?

File: ipv8/test/base.py [hunk 3]
```diff
@@ -128,41 +114,36 @@ def add_node_to_experiment(self, node):
...
-    @inlineCallbacks
-    def deliver_messages(self, timeout=.1):
+    async def deliver_messages(self, timeout=.1):
         """
         Allow peers to communicate.
 
         The strategy is as follows:
-         1. Measure the amount of working threads in the threadpool
-         2. After 10 milliseconds, check if we are down to 0 twice in a row
+         1. Measure the amount of exists asyncio tasks
```
Comment (qstokkink) [inline ipv8/test/base.py]:
```suggestion
         1. Measure the amount of existing asyncio tasks
```

File: ipv8/test/base.py [hunk 4]
```diff
@@ -128,41 +114,36 @@ def add_node_to_experiment(self, node):
...
+         2. After 10 milliseconds, check if we are below 2 tasks twice in a row
```
Comment (qstokkink) [inline ipv8/test/base.py]: Why 2?

File: run_all_tests_trial.py [hunk 5]
```diff
@@ -1,12 +1,10 @@
-from __future__ import absolute_import
-
 import inspect
```
Comment (qstokkink) [inline run_all_tests_trial.py]: This file can be removed entirely.

File: stresstest/bootstrap_rtt.py [hunk 6]
```diff
@@ -1,9 +1,15 @@
-from __future__ import print_function
-
+import time
 from os import path
 from random import randint
 from socket import gethostbyname
-import time
+
+from ipv8.community import Community, _DEFAULT_ADDRESSES, _DNS_ADDRESSES
```
Comment (qstokkink) [inline stresstest/bootstrap_rtt.py]: This is wrong, the `ipv8.*` imports should be after the `try .. except ..` that ensures that `ipv8.*` imports work.

Comment (qstokkink) [reply stresstest/bootstrap_rtt.py]: (There are more instances of this, but I'll keep it at this single comment)

File: ipv8/attestation/wallet/irmaexact/enroll_script.py (hunk 2 above)
Comment (egbertbouman) [reply ipv8/attestation/wallet/irmaexact/enroll_script.py]: Oops!

File: stresstest/bootstrap_rtt.py (hunk 6 above)
Comment (egbertbouman) [reply stresstest/bootstrap_rtt.py]: I think isort must have "fixed" this for me :disappointed:

**Comment by qstokkink** [comment]
Almost forgot: the doc snippets also need to be changed (mostly the hello world example absolutely needs to work https://github.com/Tribler/py-ipv8/blob/master/doc/basics/overlay_tutorial.rst)

**Comment by egbertbouman** [comment]
@qstokkink Just updated the docs.

**Comment by qstokkink** [comment]
I rebased this to the `asyncio` branch. I'll make sure the PR tests use Python 3 for that branch. Please leave this PR open for a bit while I make that change.

In the mean time, could you clean up your commit history?

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
The Windows machine doesn't have the requirements installed for Python 3 (`aiohttp`).

**Comment by qstokkink** [comment]
Failure on Linux:

```python
Test if the TaskManager does not allow new tasks after shutdown(). ... ok

======================================================================
FAIL: Check if data is correctly exited.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/.local/lib/python3.7/site-packages/asynctest/case.py", line 297, in run
    self._run_test_method(testMethod)
  File "/home/jenkins/.local/lib/python3.7/site-packages/asynctest/case.py", line 354, in _run_test_method
    self.loop.run_until_complete(result)
  File "/home/jenkins/.local/lib/python3.7/site-packages/asynctest/case.py", line 224, in wrapper
    return method(*args, **kwargs)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "/home/jenkins/ci/workspace/ipv8/test_ipv8_PR_unittests_linux_py36/ipv8/test/messaging/anonymization/test_community.py", line 256, in test_tunnel_data
    self.assertEqual(len(ep_listener.received_packets), 1)
AssertionError: 2 != 1
-------------------- >> begin captured logging << --------------------
asyncio: DEBUG: Using selector: EpollSelector
TunnelCommunity: DEBUG: Launching TunnelCommunity with prefix b'000281ded07332bdc775aa5a46f96de9f8f390bbc9f3'.
TunnelCommunity: INFO: Setting exitnode = 0
TunnelCommunity: DEBUG: Launching TunnelCommunity with prefix b'000281ded07332bdc775aa5a46f96de9f8f390bbc9f3'.
TunnelCommunity: INFO: Setting exitnode = 0
UDPEndpoint: DEBUG: Listening at 8080
TunnelCommunity: INFO: Want 1 data circuits of length 1
TunnelCommunity: INFO: Creating a new circuit of length 1 (type: DATA)
TunnelCommunity: INFO: First hop is required exit
TunnelCommunity: INFO: Adding first hop 211.105.60.124:26996 to circuit 2473551069
RequestCache: DEBUG: add <RetryRequestCache retry-2473551069>
TunnelCommunity: DEBUG: Got cell(2) from circuit 2473551069 (sender ('65.85.213.120', 51355), receiver Peer<211.105.60.124:26996, yLhLY5VhNjUlsl+wKuQfpVXYqYA=>)
TunnelCommunity: INFO: We joined circuit 2473551069 with neighbour ('65.85.213.120', 51355)
RequestCache: DEBUG: add <CreatedRequestCache created-2473551069>
TunnelCommunity: DEBUG: Got cell(3) from circuit 2473551069 (sender ('211.105.60.124', 26996), receiver Peer<65.85.213.120:51355, VwtBn9khepqMZ55ZL577CZtLHQ8=>)
TunnelCommunity: DEBUG: Got cell(0) from circuit 2473551069 (sender ('65.85.213.120', 51355), receiver Peer<211.105.60.124:26996, yLhLY5VhNjUlsl+wKuQfpVXYqYA=>)
TunnelCommunity: DEBUG: Got data (2473551069) from ('65.85.213.120', 51355)
TunnelCommunity: DEBUG: Data for circuit 2473551069 exiting tunnel (('localhost', 8080))
TunnelExitSocket: DEBUG: Resolved hostname localhost to ip_address 127.0.0.1
TunnelCommunity: INFO: Removing DATA circuit 2473551069 unload
TunnelCommunity: INFO: destroy_circuit 2473551069 ('211.105.60.124', 26996)
TunnelCommunity: INFO: Removed circuit 2473551069 unload
RequestCache: DEBUG: Clearing <test_ipv8_PR_unittests_linux_py36.ipv8.requestcache.RequestCache object at 0x7f1be04d6a58> [0]
TunnelCommunity: INFO: Destroy_exit_socket 2473551069 ('65.85.213.120', 51355)
RequestCache: DEBUG: Clearing <test_ipv8_PR_unittests_linux_py36.ipv8.requestcache.RequestCache object at 0x7f1be04de6a0> [1]
TunnelCommunity: INFO: Removing exit socket 2473551069 unload
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 490 tests in 77.420s

FAILED (SKIP=2, failures=1)
```

**Comment by qstokkink** [comment]
Alright, the PR tester should now detect the PR to the `asyncio` branch and run the Python 3 tests.

Steps before merge:

 - [ ] Fix [the error](https://github.com/Tribler/py-ipv8/pull/598#issuecomment-552113273)
 - [ ] Clean up the commit history
 - [ ] Install [aiohttp] on the Windows machine
 - [ ] Make sure this PR passes the tests (except the Flake 8 test)

**Comment by egbertbouman** [comment]
retest this please

**Comment by egbertbouman** [comment]
retest this please

**Comment by egbertbouman** [comment]
retest this please

**Comment by qstokkink** [comment]
@egbertbouman would you prefer to move the previously posted error to a separate issue and merge this?

**Comment by egbertbouman** [comment]
@qstokkink I've tried to reproduce the error by running the tunnel tests in a loop on an Ubuntu VM. Even after 100+ attempts, I didn't get a single failure. Since we are currently running the Linux unit tests twice and the tunnel tests use a hard-coded port number, it's probably interference between the two jobs. I've updated the tests to select a random free port number, which will hopefully fix the issue.

**Comment by qstokkink** [comment]
Good point, merge whenever you're ready.

**Pull request merged**